Riot/Modules/MatrixKit/Models/Account/MXKAccount.m

```diff
@@ -2133,28 +2133,38 @@ - (void)buildSyncFilter:(void (^)(MXFilterJSONModel *syncFilter))completion
     // Check settings
     BOOL syncWithLazyLoadOfRoomMembersSetting = [MXKAppSettings standardAppSettings].syncWithLazyLoadOfRoomMembers;
 
-    if (syncWithLazyLoadOfRoomMembersSetting)
-    {
-        // Check if the server supports LL sync filter before enabling it
-        [self supportLazyLoadOfRoomMembers:^(BOOL supportLazyLoadOfRoomMembers) {
+    void(^buildSyncFilter)(BOOL) = ^(BOOL supportsNotificationsForThreads) {
+        if (syncWithLazyLoadOfRoomMembersSetting)
+        {
+            // Check if the server supports LL sync filter before enabling it
+            [self supportLazyLoadOfRoomMembers:^(BOOL supportLazyLoadOfRoomMembers) {
 
-            if (supportLazyLoadOfRoomMembers)
-            {
-                completion([self syncFilterWithLazyLoadOfRoomMembers:YES]);
-            }
-            else
-            {
-                // No support from the HS
-                // Disable the setting. That will avoid to make a request at every startup
-                [MXKAppSettings standardAppSettings].syncWithLazyLoadOfRoomMembers = NO;
-                completion([self syncFilterWithLazyLoadOfRoomMembers:NO]);
-            }
-        }];
-    }
-    else
-    {
-        completion([self syncFilterWithLazyLoadOfRoomMembers:NO]);
-    }
+                if (supportLazyLoadOfRoomMembers)
+                {
+                    completion([self syncFilterWithLazyLoadOfRoomMembers:YES
+                                         supportsNotificationsForThreads:supportsNotificationsForThreads]);
+                }
+                else
+                {
+                    // No support from the HS
+                    // Disable the setting. That will avoid to make a request at every startup
+                    [MXKAppSettings standardAppSettings].syncWithLazyLoadOfRoomMembers = NO;
+                    completion([self syncFilterWithLazyLoadOfRoomMembers:NO
+                                         supportsNotificationsForThreads:supportsNotificationsForThreads]);
+                }
+            }];
+        }
+        else
+        {
+            completion([self syncFilterWithLazyLoadOfRoomMembers:NO supportsNotificationsForThreads:supportsNotificationsForThreads]);
+        }
+    };
+
+    [mxSession supportedMatrixVersions:^(MXMatrixVersions *matrixVersions) {
```

Comment:
It looks to me like `[self supportLazyLoadOfRoomMembers]` called above is also doing `mxSession supportedMatrixVersions]` check so we may be doing it twice? Could thse be somehow combined to get all server capabilities at once?